### ISSUE TITLE
Accessibility finding: Mishandled focus in profile edit-as-a-subtask flows

### ISSUE THREAD

**Issue by emisparkles** [department-of-veterans-affairs/va.gov-team#133956]
### Checklist item

WEB-322	Interacting with form fields doesn't trigger unexpected changes

### Severity level

2, High. Should be fixed in 1-2 sprints post-launch.

### Page(s)/url(s)

Scheduling preferences > What's the best way to contact you to schedule your appointments?
https://staging.va.gov/profile/health-care-settings/scheduling-preferences/contact-method

Scheduling preferences > When do you want us to contact you to schedule your appointments?
https://staging.va.gov/profile/health-care-settings/scheduling-preferences/contact-times

Scheduling preferences > When do you want to go to your appointments?
https://staging.va.gov/profile/health-care-settings/scheduling-preferences/appointment-times

Email and text notifications > Update your email address
https://staging.va.gov/profile/edit?fieldName=email&returnPath=%2Fprofile%2Fnotifications

Email and text notifications > Update your mobile phone number
https://staging.va.gov/profile/edit?fieldName=mobilePhone&returnPath=%2Fprofile%2Fnotifications

### Details

There is a focus bug in the scheduling preferences and notifications page edit-as-a-subtask flow. When the user interacts with a form field, focus is lost. This can be frustrating to keyboard users, especially on the appointment/contact times pages where the user may need to select multiple options. Users must keep tabbing back down to the form from the page heading.

https://github.com/user-attachments/assets/e30a697b-3ccf-4fcd-b2e1-d122d6973e67

https://github.com/user-attachments/assets/5e2f71a3-7e4f-4ad4-aec7-d6974c758998

https://github.com/user-attachments/assets/0da33911-8cca-4e6a-8179-0e597e523128

### Reproduction steps

1. Log into staging.va.gov
2. Navigate to https://staging.va.gov/profile/health-care-settings/scheduling-preferences
3. Click Edit for "What's the best way to contact you to schedule your appointments?"
4. Using a keyboard, tab into the contact method dropdown and select an option
5. Observe focus jumps to `body.merger`
6. Return to the main scheduling preferences page
7. Click Edit for "When do you want us to contact you to schedule your appointments?" or "When do you want to go to your appointments?"
8. Using a keyboard, tab to the radio form and select "No preference"
9. Observe focus jumps back to `body.merger`
10. Tab back into the form and select "Select days and times to be contacted"
11. Focus will jump back to `body.merger`
12. Continue to the next page
13. Tab into the form and select an option
14. Focus will jump back to `body.merger`
15. Navigate to https://staging.va.gov/profile/notifications
16. Click "Update your email address" or "Update your mobile phone number"
17. Tab into the text input field
18. Begin typing
19. Focus will jump to `body.merger`, inline validation error will appear

### Proposed solution or next steps

Prevent focus from getting lost when the user is interacting with forms.

Issue is likely within [this `useEffect` in the Edit.jsx file](https://github.com/department-of-veterans-affairs/vets-website/blob/7a3622cdee9b2c86a33a38ef42732cc52cde92eb/src/applications/personalization/profile/components/edit/Edit.jsx#L152-L167). Consider memoizing `fieldInfo` or using state management to ensure initial focus is only run at first render.
